daemon/cmd/daemon.go

```diff
@@ -400,8 +400,11 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 	d.endpointManager.InitMetrics()
 
 	d.redirectPolicyManager = redirectpolicy.NewRedirectPolicyManager(d.svc)
-	if option.Config.BGPAnnounceLBIP {
-		d.bgpSpeaker = speaker.New()
+	if option.Config.BGPAnnounceLBIP || option.Config.BGPAnnouncePodCIDR {
```

Comment:
@christarazi 

If we keep the BGP configuratoin flat like this, this or statement is going to continue to grow. 

This maybe out of scope for this PR, but should bgp config in daemon become structured? 

```
type BGP struct {
  AnnounceLBIP  bool
  AnnouncePodCIDR bool
}
```

And this struct gets attached to the daemon config object. Thoughts?

Reply:
In general yes I think that's a good suggestion. However, I don't feel too strongly about it because we know that we only have 3 use cases for BGP--LB IP, PodCIDR, and Static Egress IP. So in the end, the refactoring doesn't have that much of an impact, IMO.

Reply:
I guess I can see that you might have taken inspiration from `speaker.Opts`. If you do go ahead with the refactoring, then I'd be fine with it. At least the code structure is cleaned up a bit :+1:

Reply:
noted: low hanging fruit so I will handle the rest of the PR first and keep this refactor in mind.